detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/parse/GraphParserTransformer.java

```diff
@@ -53,7 +66,26 @@ private String getNameFromNode(GraphNode graphNode) {
     }
 
     private String getVersionFromLabel(String label) {
-        String[] components = label.split("\\\\n:|\\\\n");
+        String[] components = getLabelParts(label);
         return components[1];
     }
+
+    private Optional<String> getLayerFromLabel(String label, Set<String> knownLayerNames) {
+        String[] components = getLabelParts(label);
+        if (components.length == 3) {
+            String bbPath = components[2];
+            for (String candidateLayerName : knownLayerNames) {
+                String possibleLayerPathSubstring = LABEL_PATH_SEPARATOR + candidateLayerName + LABEL_PATH_SEPARATOR;
+                if (bbPath.contains(possibleLayerPathSubstring)) {
+                    return Optional.of(candidateLayerName);
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    @NotNull
+    private String[] getLabelParts(final String label) {
+        return label.split("\\\\n:|\\\\n");
```

Comment:
This seems inappropriate in a transformer. What you could do is split out this, make a LabelParser.parseLayerLabel that returns LayerDetails(path) with Optional<String> getPath() . This would let you get away from handling the String[] in other methods and let you unit test label parsing. I'd suggets both getLayerFromLabel and getLabelParts become LabelParser.parseLayer -> LayerDetails.

Reply:
Done